.github/workflows/build.yml

```diff
@@ -0,0 +1,35 @@
+name: Build MultiSense ROS
+
+on:
+  push:
+    branches:
+      - master
+  release:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: ${{ github.ref_name }}-${{ matrix.ros_distro }}-build
+      cancel-in-progress: ${{ github.ref_name != 'main' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        ros_distro:
+          - noetic
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
```

Comment:
Can all of the submodules be checked out without a token?

Reply:
yep, this is a public repo. the only submodule is libmultisense, so I can remove the recursive too